MacOS/Python/Install.sh

```diff
@@ -62,13 +61,73 @@ _py_version=$PYTHON_VERSION_PS
 # Install miniconda
 # Check if miniconda is installed
 
-echo "Installing Miniconda..."
+echo "Checking for Miniconda or Anaconda installation..."
 if conda --version > /dev/null; then
-  echo "Miniconda or anaconda is already installed"
-else
-  brew install --cask miniconda
-  [ $? -ne 0 ] && exit_message
+    echo "Miniconda or Anaconda is already installed."
+
+    # function to handle user confirmation
+    get_user_confirmation() {
+        read -p "Do you want to uninstall it? (yes/no): " confirm
+        confirm=$(echo "$confirm" | tr '[:upper:]' '[:lower:]')
+        if [[ "$confirm" != "yes" && "$confirm" != "no" ]]; then
+            echo "Please answer yes or no."
+            get_user_confirmation
+        fi
+    }
+
+    # checks for homebrew installations 
+    if [[ $(conda info --base 2> /dev/null) == *"/Caskroom/miniconda"* ]]; then
```

Comment:
instead of calling `conda info --base ` several times, do it once, store in a variable, and check that variable.

```shell
conda_info_base=$(conda info --base 2> /dev/null)
```